crates/api_common/src/site.rs

```diff
@@ -648,15 +642,29 @@ pub struct GetUnreadRegistrationApplicationCountResponse {
 #[cfg_attr(feature = "full", derive(TS))]
 #[cfg_attr(feature = "full", ts(export))]
 /// Block an instance as user
-pub struct BlockInstance {
+pub struct UserBlockInstanceParams {
   pub instance_id: InstanceId,
   pub block: bool,
 }
 
-#[skip_serializing_none]
-#[derive(Debug, Serialize, Deserialize, Clone)]
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "full", derive(TS))]
+#[cfg_attr(feature = "full", ts(export))]
+pub struct AdminBlockInstanceParams {
+  pub instance: String,
+  pub block: bool,
+  #[cfg_attr(feature = "full", ts(optional))]
+  pub reason: Option<String>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  pub expires: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "full", derive(TS))]
 #[cfg_attr(feature = "full", ts(export))]
-pub struct BlockInstanceResponse {
-  pub blocked: bool,
+pub struct AdminAllowInstanceParams {
+  pub instance: String,
```

Comment:
Had to change these from instance_id to domain, otherwise it would be very complicated to initialize the tests. It would also be possible to allow both instance id or domain in the same field, but ts-rs would probably not understand that.